ddtrace/llmobs/_integrations/anthropic.py

```diff
@@ -0,0 +1,147 @@
+import json
+from typing import Any
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+
+from ddtrace._trace.span import Span
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.llmobs._constants import INPUT_MESSAGES
+from ddtrace.llmobs._constants import METADATA
+from ddtrace.llmobs._constants import MODEL_NAME
+from ddtrace.llmobs._constants import OUTPUT_MESSAGES
+from ddtrace.llmobs._constants import SPAN_KIND
+
+from .base import BaseLLMIntegration
+
+
+log = get_logger(__name__)
+
+
+API_KEY = "anthropic.request.api_key"
+MODEL = "anthropic.request.model"
+TOTAL_COST = "anthropic.tokens.total_cost"
+TYPE = "anthropic.request.type"
+
+
+class AnthropicIntegration(BaseLLMIntegration):
+    _integration_name = "anthropic"
+
+    def llmobs_set_tags(
+        self,
+        resp: Any,
+        span: Span,
+        args: List[Any],
+        kwargs: Dict[str, Any],
+        err: Optional[Any] = None,
+    ) -> None:
+        """Extract prompt/response tags from a completion and set them as temporary "_ml_obs.*" tags."""
+        # if not self.llmobs_enabled:
+        #     return
+        parameters = {
+            "temperature": float(span.get_tag("anthropic.request.parameters.temperature") or 1.0),
+            "max_tokens": int(span.get_tag("anthropic.request.parameters.max_tokens") or 0),
+        }
+        messages = get_argument_value(args, kwargs, 0, "messages")
+        input_messages = self._extract_input_message(messages)
+
+        span.set_tag_str(SPAN_KIND, "llm")
+        span.set_tag_str(MODEL_NAME, span.get_tag("anthropic.request.model") or "")
+        span.set_tag_str(INPUT_MESSAGES, json.dumps(input_messages))
+        span.set_tag_str(METADATA, json.dumps(parameters))
+        if err or resp is None:
+            span.set_tag_str(OUTPUT_MESSAGES, json.dumps([{"content": ""}]))
+        else:
+            output_messages = self._extract_output_message(resp)
+            span.set_tag_str(OUTPUT_MESSAGES, json.dumps(output_messages))
+
+        usage = _get_attr(resp, "usage")
+        self.record_usage(
+            span,
+            {"prompt": _get_attr(usage, "input_tokens", 0), "completion": getattr(usage, "output_tokens", 0)},
+        )
```

Comment:
`AnthropicIntegration.record_usage()` should be used to tag span metrics for the input/output/total token counts. We should have a separate `AnthropicIntegration._set_llmobs_metrics_tags()` to return the recorded span metric values, i.e.
```python
def _get_llmobs_metrics_tags(span):
    return {"input_tokens": span.get_metric("anthropic.response.usage.input_tokens"), "output_tokens": ..., "total_tokens": ...}
```
And set that on the span, i.e. `span.set_tag_str(METRICS, json.dumps(self._get_llmobs_metrics_tags(span))`

Reply:
and we set this tag in the LLMObs Integration correct?